src/client.rs

```diff
@@ -0,0 +1,152 @@
+use std::{path::Path, str::FromStr, sync::Arc};
+
+pub use hopr_chain_connector;
+pub type HoprEdgeClient = Hopr<Arc<HoprBlockchainSafeConnector<BlokliClient>>, HoprNodeDb>;
+
+use futures::future::{AbortHandle, abortable};
+use hopr_chain_connector::{
+    blokli_client::BlokliClient,
+    {HoprBlockchainSafeConnector, init_blokli_connector},
+};
+use hopr_db_node::{HoprNodeDb, init_hopr_node_db};
+use hopr_lib::{Hopr, HoprKeys, ToHex, api::chain::ChainEvents, config::HoprLibConfig};
+use tracing::info;
+
+use crate::errors::EdgliError;
+
+pub async fn run_hopr_edge_node_with<F, T>(
+    cfg: HoprLibConfig,
+    db_data_path: &Path,
+    hopr_keys: HoprKeys,
+    f: F,
+) -> anyhow::Result<AbortHandle>
```

Comment:
The AbortHandles might be better not exposed and managed with the lifetime of the Edgli singleton (if you remove the Clone derive on `Edgli`)